### PR TITLE
Generation of public api, proguard rules and api changes

### PR DESCRIPTION
**Pull request by VovaMedvediev**
In this version of plugin generation of public api, proguard rules and api changes were implemented

### CODE REVIEW

**Comment by AleksanderLevadniy** [comment]
Please, restore files under `.idea` folder

File: build.gradle [hunk 1]
```diff
@@ -9,6 +9,7 @@ buildscript {
     repositories {
         google()
         jcenter()
+        mavenCentral()
```
Comment (AleksanderLevadniy) [inline build.gradle]: Why do we need it?

File: src/main/kotlin/com/aol/mobile/sdk/publicapiplugin/PublicApiExtension.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.aol.mobile.sdk.publicapiplugin
```
Comment (AleksanderLevadniy) [inline src/main/kotlin/com/aol/mobile/sdk/publicapiplugin/PublicApiExtension.kt]: Please, change namespace to `com.verizonmedia.mobile.publicapi` for all sources

File: src/main/kotlin/com/aol/mobile/sdk/publicapiplugin/PublicApiPlugin.kt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                        (getByName("bundle${variant.name.capitalize()}Aar") as AbstractArchiveTask).from(manifest)
+                        getByName("merge${variant.name.capitalize()}ConsumerProguardFiles").dependsOn(genProguardTask)
+                        getByName("assemble").dependsOn(checkApiTask)
+                    }
+                }
+            }
+        }
+    }
+
+    override fun apply(project: Project) {
```
Comment (AleksanderLevadniy) [inline src/main/kotlin/com/aol/mobile/sdk/publicapiplugin/PublicApiPlugin.kt]: Please, move this method to the top

**Review by AleksanderLevadniy — CHANGES_REQUESTED** [review]
Please, review comments

File: settings.gradle [hunk 4]
```diff
@@ -2,3 +2,4 @@
  * Copyright 2018, Oath Inc.
  * Licensed under the terms of the MIT License. See LICENSE.md file in project root for terms.
  */
+include ":"
```
Comment (AleksanderLevadniy) [inline settings.gradle]: This is not needed

File: src/main/kotlin/com/verizonmedia/mobile/publicapi/cilib/AndroidCiLibrary.kt [hunk 5]
```diff
@@ -107,10 +107,10 @@ class AndroidCiLibrary : Plugin<Project> {
                         minSdkVersion(minAndroidApi)
                         targetSdkVersion(targetAndroidApi)
 
-                        testInstrumentationRunner = "android.support.test.runner.AndroidJUnitRunner"
-                        versionCode = 1
-                        version = libraryVersion
-                        versionName = libraryVersion
+//                        testInstrumentationRunner = "android.support.test.runner.AndroidJUnitRunner"
```
Comment (AleksanderLevadniy) [inline src/main/kotlin/com/verizonmedia/mobile/publicapi/cilib/AndroidCiLibrary.kt]: Remove comments

File: settings.gradle [hunk 6]
```diff
@@ -2,3 +2,4 @@
  * Copyright 2018, Oath Inc.
  * Licensed under the terms of the MIT License. See LICENSE.md file in project root for terms.
  */
+include ":pluginnn"
```
Comment (AleksanderLevadniy) [inline settings.gradle]: Why do we need this?

**Review by AleksanderLevadniy — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/aol/mobile/sdk/cilib/AndroidCiLibrary.kt [hunk 7]
```diff
@@ -107,10 +108,10 @@ class AndroidCiLibrary : Plugin<Project> {
                         minSdkVersion(minAndroidApi)
                         targetSdkVersion(targetAndroidApi)
 
-                        testInstrumentationRunner = "android.support.test.runner.AndroidJUnitRunner"
-                        versionCode = 1
-                        version = libraryVersion
-                        versionName = libraryVersion
+//                        testInstrumentationRunner = "android.support.test.runner.AndroidJUnitRunner"
```
Comment (AleksanderLevadniy) [inline src/main/kotlin/com/aol/mobile/sdk/cilib/AndroidCiLibrary.kt]: Please, remove comments

File: src/test/kotlin/com/aol/mobile/sdk/apitracker/DescriptorsDeserializerTest.kt [hunk 8]
```diff
@@ -21,7 +21,7 @@ class DescriptorsDeserializerTest {
                   "public",
                   "final"
                 ],
-                "name": "com.aol.mobile.sdk.player.OneSDK",
```
Comment (AleksanderLevadniy) [inline src/test/kotlin/com/aol/mobile/sdk/apitracker/DescriptorsDeserializerTest.kt:24]: Why this file is changed?

Comment (VovaMedvediev) [reply src/test/kotlin/com/aol/mobile/sdk/apitracker/DescriptorsDeserializerTest.kt:24]: > Why this file is changed?

This file has links to classes which namespace was changed.

**Review by AleksanderLevadniy — APPROVED** [review]
👍

**Pull request closed without merging**